code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -25,12 +25,98 @@
 
 	if(iscarbon(L))
 		var/mob/living/carbon/C = L
-		if(C.get_blood_id() == /datum/reagent/blood && (method == INJECT || (method == INGEST && C.dna && C.dna.species && (DRINKSBLOOD in C.dna.species.species_traits))))
-			if(!data || !(data["blood_type"] in get_safe_blood(C.dna.blood_type)))
+		if(C.get_blood_id() == /datum/reagent/blood || C.get_blood_id() == /datum/reagent/blood/true_draculine && (method == INJECT || (method == INGEST && C.dna && C.dna.species && (DRINKSBLOOD in C.dna.species.species_traits))))
+			if((!data || !(data["blood_type"] in get_safe_blood(C.dna.blood_type))) && C.get_blood_id() != /datum/reagent/blood/true_draculine)
 				C.reagents.add_reagent(/datum/reagent/toxin, reac_volume * 0.5)
 			else
 				C.blood_volume = min(C.blood_volume + round(reac_volume, 0.1), BLOOD_VOLUME_MAXIMUM)
+/datum/reagent/blood/true_draculine
+	name = "True Draculine"
+	data = list("donor"=null,"viruses"=null,"blood_DNA"=null,"blood_type"="Draculine","resistances"=null,"trace_chem"=null,"mind"=null,"ckey"=null,"gender"=null,"real_name"=null,"cloneable"=null,"factions"=null,"quirks"=null)
+	description = "Slowly heals all damage types. Overdose will, after a short while, turn you into a vampire, addiction lowers your max health as your body attempts to fight off the corruption, if you're incompatible you're immune to the addiction and overdose will damage you instead."
+	metabolization_rate = 1
+	addiction_threshold = 20
+	overdose_threshold = 30
+	var/healing = 0.20
+
+/datum/reagent/blood/true_draculine/on_mob_add(mob/living/carbon/human/M)
+	if(M.dna.species.exotic_blood)
+		src.addiction_threshold = 0
+	if(M.dna.species.exotic_blood == /datum/reagent/blood/true_draculine)
+		src.overdose_threshold = 0
+	..()
+
+/datum/reagent/blood/true_draculine/on_mob_life(mob/living/carbon/human/M)
+	if(M.dna.species.exotic_blood != /datum/reagent/blood/true_draculine)
+		M.adjustToxLoss(-healing*REM, 0, TRUE)
+		M.adjustOxyLoss(-healing*REM, 0)
+		M.adjustBruteLoss(-healing*REM, 0)
+		M.adjustFireLoss(-healing*REM, 0)
+		M.Jitter(2)
+		. = 1
+	..()
+
+/datum/reagent/blood/true_draculine/overdose_process(mob/living/carbon/human/M)
+	if(M.dna.species.exotic_blood)
+		M.adjustToxLoss(1*REM, 0)
+		M.adjustOxyLoss(1*REM, 0)
+		M.adjustBruteLoss(1*REM, FALSE, FALSE, BODYPART_ORGANIC)
+		M.adjustFireLoss(1*REM, FALSE, FALSE, BODYPART_ORGANIC)
+		M.Jitter(6)
+	else
+		if(current_cycle >= 50)
+			for(var/datum/reagent/addiction in M.reagents.addiction_list)
+				if(addiction.name == "True Draculine")
+					M.reagents.remove_addiction(addiction)
+			M.add_quirk(/datum/quirk/vampire)
+	..()
+	. = 1
 
+/datum/reagent/blood/true_draculine/addiction_act_stage1(mob/living/carbon/human/M)
+	M.dna.species.species_traits |= list(DRINKSBLOOD)
```

Comment:
Should be an actual trait, I think. See #458 for an example.

Reply:
It's like this just because that's how drinksblood already operates, processing blood specifically checks for the flag DRINKSBLOOD it in species_traits rather than a trait proper but I could make the changes later